includes/handler/class-inbox.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Inbox handler file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Handler;
+
+use Activitypub\Activity\Activity;
+use Activitypub\Activity\Base_Object;
+use Activitypub\Collection\Inbox as Inbox_Collection;
+
+/**
+ * Handle Inbox requests.
+ */
+class Inbox {
+	/**
+	 * Initialize the class, registering WordPress hooks.
+	 */
+	public static function init() {
+		// Check if inbox collection persistence is enabled.
+		if ( \get_option( 'activitypub_persist_inbox', '0' ) ) {
+			\add_action(
+				'activitypub_inbox',
+				array( self::class, 'handle_inbox_requests' ),
+				10,
+				4
+			);
+		}
+	}
+
+	/**
+	 * Handles "Inbox" requests.
+	 *
+	 * @param array              $data     The data array.
+	 * @param int                $user_id  The id of the local blog-user.
+	 * @param string             $type     The type of the activity.
+	 * @param Activity|\WP_Error $activity The Activity object.
+	 */
+	public static function handle_inbox_requests( $data, $user_id, $type, $activity ) {
+		/**
+		 * Filters the activity types to persist in the inbox.
+		 *
+		 * @param array $activity_types The activity types to persist in the inbox.
+		 */
+		$activity_types = \apply_filters( 'activitypub_persist_inbox_activity_types', array( 'Create', 'Update' ) );
```

Comment:
Could you remind me why we don't use the type-specific handlers for this? I don't remember where we left off on that, if at all.

Reply:
the inbox should persist all incoming activities, this limitation is only for the time beeing to not flood the inbox in the testing phase. If I would use the type-specific ones, I would have to implement the storing in every class and would have to add even more.